### ISSUE TITLE
(feature) Event props on components

### ISSUE THREAD

**Issue by necolas** [facebook/react-strict-dom#38]
### Describe the feature request

Support web standard event props on all components. This element is not currently supported on native. Although some event props exist in React Native, most of the events are not fired synchronously and/or at the right time, and the event objects themselves do not match web standards (apart from the new PointerEvent implementation).

* onAuxClick
* onBeforeInput (input)
* onBeforeInput (select)
* onBeforeInput (textarea)
* onBlur
* onChange (input)
* onChange (select)
* onChange (textarea)
* onClick
* onContextMenu
* onCopy
* onCut
* onError (img)
* onFocus
* onGotPointerCapture
* onInput (input)
* onInput (select)
* onInput (textarea)
* onInvalid (input)
* onInvalid (select)
* onInvalid (textarea)
* onKeyDown
* onKeyUp
* onLoad (img)
* onLostPointerCapture
* onPaste
* onPointerCancel
* onPointerDown
* onPointerEnter
* onPointerLeave
* onPointerMove
* onPointerOut
* onPointerOver
* onPointerUp
* onSelect
* onSelectionChange

One way to approach this could be to introduce new, synchronous event props in React Native (e.g., `onKeyDownSync`) that do not require changes to the existing event props. There has been interest in doing this for existing RN use cases beyond web compability. RSD could then use these new props (and rename them in the user-space layer for now) to provide events and event objects that match web.

**Comment by robik** [comment]
Hey, I'm currently researching the topic and working on some kind of React Native RFC and I wonder if this issue scope also covers the event bubbling and propagation, or is it a different scope? As in, should we be able to listen on `onPressCapture` on arbitrary `<View>` and have possibility to stop propagating it (e.g. to disable `onClick` in that view and children)?

**Comment by rubennorte** [comment]
> Hey, I'm currently researching the topic and working on some kind of React Native RFC and I wonder if this issue scope also covers the event bubbling and propagation, or is it a different scope? As in, should we be able to listen on `onPressCapture` on arbitrary `<View>` and have possibility to stop propagating it (e.g. to disable `onClick` in that view and children)?

I think that's also in scope but it might take a while to align how events work in RN and on Web.

**Comment by necolas** [comment]
> I wonder if this issue scope also covers the event bubbling and propagation

Yes definitely. The events have to match W3C events as exposed in React DOM - they need to be sync, dispatched in the expected sequence, and the payload has to match the W3C events (then wrapped by React's event system). In the main RFC, there is an overview. The event prop names there are a bit different to what we'll probably end up using (`on-click` vs `onClick`), because we'll probably not attempt to undo React DOM's event wrappers at this stage. In React Native, I expect the new props will have names like `onClickSync` to differentiate from any existing event props, which will we do not need to change the semantics of. https://github.com/necolas/discussions-and-proposals/blob/reduce-fragmentation/proposals/0000-reduce-fragmentation.md#common-events

In addition to the event props there is the need to support the EventTarget API on host nodes; another way to listen to events and supports custom event dispatching https://github.com/necolas/discussions-and-proposals/blob/reduce-fragmentation/proposals/0000-reduce-fragmentation.md#eventtarget

**Comment by CAMOBAP** [comment]
In vanilla react-native `onPointer*` disabled for both:
- iOS: `RCTDispatchW3CPointerEvents = NO` - [RCTConstants.m](https://github.com/react/react-native/blob/6b45e579d6c17e2ff43709812810b4a32ec81a40/packages/react-native/React/Base/RCTConstants.m#L31)
- Android: `dispatchPointerEvents = false` - [ReactFeatureFlags.kt](https://github.com/react/react-native/blob/6b45e579d6c17e2ff43709812810b4a32ec81a40/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/config/ReactFeatureFlags.kt#L23)

Is this still something planned to do?
